subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/AbstractInstalledToolChainIntegrationSpec.groovy

```diff
@@ -39,16 +41,35 @@ abstract class AbstractInstalledToolChainIntegrationSpec extends AbstractIntegra
 
     def setup() {
         initScript = file("init.gradle") << """
-allprojects { p ->
-    apply plugin: ${toolChain.pluginClass}
-
-    model {
-          toolChains {
-            ${toolChain.buildScriptConfig}
-          }
-    }
-}
-"""
+            allprojects { p ->
+                apply plugin: ${toolChain.pluginClass}
+            
+                model {
+                      toolChains {
+                        ${toolChain.buildScriptConfig}
+                      }
+                }
+            }
+        """
+        if (toolChain.meets(ToolChainRequirement.WINDOWS_GCC)) {
+            initScript << """
+                allprojects { p ->
+                    [
+                       "application": ["cpp-application", "swift-application"],
+                        "library": ["cpp-library", "swift-library"],
+                        "unitTest": ["cpp-unit-test"]
+                    ].each { block, plugins ->
```

Comment:
as another alternative...

Could this do something like this instead of looking particular plugins/components?
```
components.withType(CppComponent) {
   // configure targetMachines
}
```
I don't think we would ever run Swift tests with WINDOWS_GCC?

Reply:
Yeah, that seems a lot simpler, doesn't it...